src/components/Table/Table.tsx

```diff
@@ -0,0 +1,104 @@
+import { Icon } from "@/components";
+import { IconName } from "@/components/Icon/types";
+import styled from "styled-components";
+
+type TableHeaderProps = {
+  children: React.ReactNode;
+  leftIcon?: IconName;
+  rightIcon?: IconName;
+};
+const StyledHeader = styled.th`
+  padding: 8px 16px;
+  background-color: ${props =>
+    props.theme.click.table.header.color.background.default};
+  font: ${props => props.theme.click.table.header.title.default};
+  color: ${props => props.theme.click.table.header.color.title.default};
+  gap: 4px;
+  text-align: left;
+
+  &:first-of-type {
+    border-top-left-radius: ${props => props.theme.click.table.radii.all};
+  }
+  &:last-of-type {
+    border-top-right-radius: ${props => props.theme.click.table.radii.all};
+  }
+`;
+
+const HeaderContentWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: start;
+  gap: 4px;
+`;
+
+const TableHeader = ({
+  leftIcon,
+  rightIcon,
+  children,
+  ...delegated
+}: TableHeaderProps) => (
+  <StyledHeader {...delegated}>
+    <HeaderContentWrapper>
+      {leftIcon && <Icon name={leftIcon} width="12px" height="12px" />}
+      {children}
+      {rightIcon && <Icon name={rightIcon} width="12px" height="12px" />}
+    </HeaderContentWrapper>
+  </StyledHeader>
+);
+
+const TableRow = styled.tr`
```

Comment:
we should add a name to the generated component so that on story book docs it does not show up as `styled.tr`

Reply:
done